src/User/UserRepository.php

```diff
@@ -75,7 +75,7 @@ public function findByEmail($email)
      */
     public function getIdForUsername($username, User $actor = null)
     {
-        $query = User::where('username', 'like', $username);
+        $query = User::where('username', $username);
```

Comment:
Right now, we do a case-insensitive search here - this would change that. Is that intentional? At least for user search, case-insensitivity would probably be nice, right?

Reply:
This method is used on profile pages, these url's are generated by the app and should already match the username precisely. Nevertheless I agree that an exact lowercase query would be better: how about `lower(username) = $username` and use strtolower on that var beforehand?

Reply:
Yes, it's (probably) fine for those pages, but the method is also used in user searches (`AuthorGambit`). Your solution sounds fine, but it may be more complex when we tackle UTF-8 usernames / display names.